### DISCUSSION TITLE
q-uploader - if image, resize and compress it before upload - v1

### DISCUSSION THREAD

**Discussion by DocBouCou** [quasarframework/quasar#13722] [General - Components / Directives / etc] [answered]
Hi,

In my case, the user can only add one file in q-uploader. 
I would like, if the file is an image, to resize and compress it (using browser-image-compression) _before_ to upload it to the server (if it is not an image, just upload it).

How would that be possible? 
Should I listen the @added event, and do checking-compression there, or should I use factory function (if so, how?).

Regarding the logic, the coding etc, and how to do that using a native or quasar file picker, that is ok for me. 

What I do not know, it's how to do that using q-uploader? Where/when should I implement the logic (in the factory? @added? @click in scoped slot?)? 
More specifically, I think that I do not understand how to manipulate the file using q-uploader before to upload it?

I hope that someone in the community could help me :)
Regards,

**Comment by metalsadman** [comment, marked as answer]
check `methods` section of the api card, you can add/remove, etc... programatically using the component's exposed methods via template ref. ie. `uploaderRef.value.addFiles(...)`

![Screen Shot 2022-06-21 at 9 34 18 AM](https://user-images.githubusercontent.com/31038838/174698059-a70889cb-329a-4fda-a47a-983c92314c3b.png)

**Reply by DocBouCou** [reply]
OK, thank you for your answer. 
I needed to know if `@added` -> modify file -> `removeFiles()` -> `addFiles()` was indeed the way to go. So, as there is not a loading prop for this component, I will have to use the slots to add loading myself (so the user cannot upload during the process, or knows that something is going on).
